server/channels/app/integration_action_test.go

```diff
@@ -209,23 +209,28 @@ func TestPostAction(t *testing.T) {
 				jsonErr := json.NewDecoder(r.Body).Decode(&request)
 				assert.NoError(t, jsonErr)
 
-				assert.Equal(t, request.UserId, th.BasicUser.Id)
-				assert.Equal(t, request.UserName, th.BasicUser.Username)
-				assert.Equal(t, request.ChannelId, channel.Id)
-				assert.Equal(t, request.ChannelName, channel.Name)
+				assert.Equal(t, th.BasicUser.Id, request.UserId)
+				assert.Equal(t, th.BasicUser.Username, request.UserName)
+				assert.Equal(t, channel.Id, request.ChannelId)
+				assert.Equal(t, channel.Name, request.ChannelName)
 				if channel.Type == model.ChannelTypeDirect || channel.Type == model.ChannelTypeGroup {
 					assert.Empty(t, request.TeamId)
 					assert.Empty(t, request.TeamName)
 				} else {
-					assert.Equal(t, request.TeamId, th.BasicTeam.Id)
-					assert.Equal(t, request.TeamName, th.BasicTeam.Name)
+					assert.Equal(t, th.BasicTeam.Id, request.TeamId)
+					assert.Equal(t, th.BasicTeam.Name, request.TeamName)
 				}
 				assert.True(t, request.TriggerId != "")
 				if request.Type == model.PostActionTypeSelect {
-					assert.Equal(t, request.DataSource, "some_source")
-					assert.Equal(t, request.Context["selected_option"], "selected")
+					if selectedOption, ok := request.Context["selected_option"]; ok {
+						// If something was selected, confirm that the data source and selected option are present
+						assert.Equal(t, "users", request.DataSource)
+						assert.Equal(t, "selected", selectedOption)
+					} else {
+						assert.Empty(t, request.DataSource)
+					}
```

Comment:
The test was actually wrong. `request.Context["selected_option"]` and `request.DataSource` are only provided if an option is selected. Updated it.